src/CycloneDX.Core/Models/CryptoProperties.cs

```diff
@@ -0,0 +1,330 @@
+            set 
+            {
+                this.Algorithms = new List<string>(value.Algorithms);
+            }
+        }
+        [XmlIgnore]
+        [ProtoMember(3)]
+        public List<string> Identifiers { get; set; }
+        [XmlElement("identifiers")]
+        [JsonIgnore]
+        public CipherSuiteIdentifierCollection Identifiers_XML
+        {
+            get
+            {
+                return new CipherSuiteIdentifierCollection { Identifiers = this.Identifiers };
+            }
+            set
+            {
+                this.Identifiers = new List<string>(value.Identifiers);
+            }
+        }
+    }
+
+    public class CipherSuiteAlgorithmCollection
+    {
+        [XmlElement("algorithm")]
+        public List<string> Algorithms { get; set; }
+    }
+    public class CipherSuiteIdentifierCollection
+    {
+        [XmlElement("identifier")]
+        public List<string> Identifiers { get; set; }
+    }
+
+
+
+    public enum AssetType
+    {
+   //     Null,
+        [XmlEnum("algorithm")]
+        Algorithm,
+        [XmlEnum("certificate")]
+        Certificate,
+        [XmlEnum("protocol")]
+        Protocol,
+        [XmlEnum("related-crypto-material")]
+        Related_Crypto_Material,
+
+    }
+
+    [ProtoContract]
+    public class RelatedCryptoMaterialProperties
+    {
+        [XmlElement("type")]
+        [ProtoMember(1)]
+        public RelatedCryptoMaterialType Type { get; set; }
+        [XmlElement("id")]
+        [ProtoMember(2)]
+        public string Id { get; set; }
+        [XmlElement("state")]
+        [ProtoMember(3)]
+        public KeyState State { get; set; }
+        [XmlElement("algorithmRef")]
+        [ProtoMember(4)]
+        public string AlgorithmRef { get; set; }
+        [XmlElement("creationDate")]
+        [ProtoMember(5)]
+        public DateTime CreationDate { get; set; }
+        [XmlElement("activationDate")]
+        [ProtoMember(6)]
+        public DateTime ActivationDate { get; set; }
+        [XmlElement("updateDate")]
+        [ProtoMember(7)]
+        public DateTime UpdateDate { get; set; }
+        [XmlElement("expirationDate")]
+        [ProtoMember(8)]
+        public DateTime ExpirationDate { get; set; }
+        [XmlElement("value")]
+        [ProtoMember(9)]
+        public string Value { get; set; }
+        [XmlElement("size")]
+        [ProtoMember(10)]
+        public int Size { get; set; }
+        [XmlElement("format")]
+        [ProtoMember(11)]
+        public string Format { get; set; }
+        [XmlElement("securedBy")]
+        [ProtoMember(12)]
+        public SecuredBy SecuredBy { get; set; }
+
+    }
+
+    [ProtoContract]
+    public class SecuredBy
+    {
+        [XmlElement("mechanism")]
+        [ProtoMember(1)]
+
+        public string Mechanism { get; set; }
+        [XmlElement("algorithmRef")]
+        [ProtoMember(2)]
+
+        public string AlgorithmRef { get; set; }
+
+
+    }
+
+
+    public enum RelatedCryptoMaterialType
```

Comment:
I think here we have an issue with the Protobuf serialization (Order as well as Null value).

Reply:
Your PR will resolve this, right?

Reply:
Yes, it should.